libopenage/rng/rng.cpp

```diff
@@ -48,58 +48,93 @@ void RNG::seed(uint64_t val) {
 
 
 /**
- * murmurhash3 bit avalancher - output is uniformly distributed
- * and changing one bit of input results in ~1/2 of output bits changed
- * This function is a good way of mixing up an input,
- * or mixing together two results with hash64(val1 ^ val2)
- *
- * https://en.wikipedia.org/wiki/MurmurHash
- */
-static uint64_t hash64(uint64_t val) {
-	// The two magic numbers have been determined heuristically.
-	// They avalanche all bits in val to within 0.25% bias.
-
-	val ^= val >> 33;
-	val *= 0xff51afd7ed558ccd;
-	val ^= val >> 33;
-	val *= 0xc4ceb9fe1a85ec53;
-	return val ^ (val >> 33);
+* Rotate left no carry
+*
+* http://blog.regehr.org/archives/1063
+*/
+static inline uint64_t rotl64(uint64_t x, uint64_t n)
```

Comment:
I just realized that there is already a rotate left function in `libopenage/util/misc.h`, I'll use that one instead